File: chpt1/section_1_4/problemSet/problem_1_4_21.m
% 1.4.21 The solution to y' = 2y + δ(t − 3) jumps up by 1 at t = 3. Before and after t = 3,
% the delta function is zero and y grows like e2t. Draw the graph of y(t) when
% (a) y(0) = 0 and (b) y(0) = 1. Write formulas for y(t) before and after t = 3.
%
% Solution to A) y(t) = H(t-3)e^(2(t-3))
%
% run with $ matlab -batch "problem_1_4_21"

t = 0.0:0.01:10.0;

y1 = heaviside(t-3).*exp(2*(t-3));
y2 = exp(2*t) + heaviside(t-3).*exp(2*(t-3));

f = figure;
plot(t, y1, t, y2);
legend("y1", "y2");

ylim([0 5]);
xlim([0 5]);
grid on;
title("1.4.21 plot of y1(t) = H(t-3)e^{2(t-3)} and y2(t) = e^{2t} + H(t-3)e^{2(t-3)}");
xlabel("time");
ylabel("y");
uiwait(f);

File: chpt1/section_1_4/problemSet/problem_1_4_21.py
import numpy as np
import matplotlib.pyplot as plt

t = np.linspace(-5, 5, 400)  # 400 points from 0 to 1
y1 = np.heaviside(t - 3, 1)*(np.exp(-2*(t - 3)))  # Compute e^{t} for each t
y2 = (np.exp(-2*(t))) + np.heaviside(t - 3, 1)*(np.exp(-2*(t - 3)))  # Compute e^{t} for each t


plt.figure(figsize=(8, 8))
plt.plot(t, y1, label='$y1 = H(t-3)*e^{-2(t-3)}$')
plt.plot(t, y2, label='$y2 = e^{-2t} + H(t-3)*e^{-2(t-3)}$')

# Adding title and labels
plt.title('Graph of $y1 = H(t-3)*e^{-2(t-3)}$ and $y2 = e^{-2t} + H(t-3)*e^{-2(t-3)}$')
plt.xlim(-1, 5)
plt.ylim(-5, 5)
plt.xlabel('t')
plt.ylabel('y')

# Add a grid and legend
plt.grid(True)
plt.legend()

# Show the plot
plt.show()